src/main/java/mate/academy/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.category.CategoryDto;
+import mate.academy.dto.category.CategoryRequestDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.CategoryMapper;
+import mate.academy.model.Category;
+import mate.academy.repository.CategoryRepository;
+import mate.academy.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository repository;
+    private final CategoryMapper mapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return repository.findAll(pageable)
+                .map(mapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        return repository.findById(id)
+                .map(mapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException(categoryErrorMessage(id)));
+    }
+
+    @Override
+    public CategoryDto save(CategoryRequestDto category) {
+        return mapper.toDto(repository.save(mapper.toCategory(category)));
```

Comment:
would make sense to declare a separate method for `toCaregory` - it will improve reabability

Reply:
you mean create separate variable? (correct me if I'm wrong)
fixed;